backend/learn/tests/test_credits.py

```diff
@@ -0,0 +1,104 @@
+from django.test import TestCase
+from learn.models import Block, Task, Level, Student, Toolbox, TaskSession
+from learn.credits import get_credits, get_earned_credits, \
+    total_credits_for_levels
+
+import pytest
+
+
+class BlockTestCase(TestCase):
+    def test_blocks_exists(self):
+        assert Block.objects.exists()
+
+    def test_blocks_are_ordered(self):
+        first_retrieved_blocks = list(Block.objects.all())[:3]
+        first_expected_blocks = [
+            Block.objects.get(name='fly'),
+            Block.objects.get(name='shoot'),
+            Block.objects.get(name='repeat')]
+        assert first_retrieved_blocks == first_expected_blocks
+
+    def test_str_returns_name(self):
+        carrot_block = Block(name='carrot', order=4)
+        assert str(carrot_block) == 'carrot'
+
+
+@pytest.fixture
+def toolbox():
+    return Toolbox(name="some toolbox")
+
+
+@pytest.fixture
+def level(toolbox):
+    return Level(level=5, name="some level", toolbox=toolbox, credits=20)
+
+
+@pytest.fixture
+def task(level):
+    return Task(name="some task", level=level, setting="", solution="")
+
+
+@pytest.fixture
+def student():
+    return Student(user=None)
+
+
+@pytest.mark.django_db
+@pytest.fixture(scope='function')
+def solved_task_for_student():
+    toolbox = Toolbox(name="some toolbox")
+    toolbox.save()
+
+    level = Level(level=5, name="some level", toolbox=toolbox, credits=20)
+    level.save()
+
+    task = Task(name="some task", level=level, setting="", solution="")
+    task.save()
+
+    student = Student(user=None)
+    student.save()
+
+    task_session = TaskSession(student=student, task=task, solved=True)
+    task_session.save()
+    return student, task
+
+
+@pytest.fixture
+def levels():
+    return [
+        Level(level=1, name="some level 1", toolbox=None, credits=10),
+        Level(level=2, name="some level 2", toolbox=None, credits=20),
+        Level(level=3, name="some level 3", toolbox=None, credits=30)
+    ]
+
+
+def test_get_credits_for_task(task):
```

Comment:
Ideally, each test should be understandable just from looking at the test. Here, you need to look up the task fixture and explore why it should have the expected number of credits as stated. So my suggestion how to improve the readability even more is to create a helper function/fixture to get task at a certain level, because the level is the only thing important about the task to determine the level. The test would look like this:

```
task = create_task_at_level(5)
credits = get_credits(task)
expected_credits = (5 + 1) ** 2
assert expected_credits == credits
```

It would also allow for a parametrized test (testing e.g. levels 0, 1, and 5).

What do you think? Is it possible, or are there some issues with this approach?

(Same for some other tests below.)